### PR TITLE
Migrate jira

### PR DESCRIPTION
**Pull request by baconYao**
In this commit, I did the following things.

- Migrate the whole Jira API business logic
- Refactor a small peice of code of base.py and restructure the project
- Add unit test for base.py

### TBD

- Refactor the PC scenario for better maintaince
- Add unit test for PC scenario
- Complete the README files

### CODE REVIEW

**Comment by baconYao** [comment]
### Test result
```bash
$ python -m unittest -v tests.test_base_api

test_api_property (tests.test_base_api.JiraAPITest)
Check the properties of JiraAPI class are correct ... ok
test_create_jira_fields_template (tests.test_base_api.JiraAPITest)
Check the most basic part of card's content is correct with those given ids ... ok
test_create_paragraph_content (tests.test_base_api.JiraAPITest)
Check the format of paragraph matches the expected result ... ok
test_create_table_content (tests.test_base_api.JiraAPITest)
Check the format of table matches the expected result ... ok

----------------------------------------------------------------------
Ran 4 tests in 0.001s

OK
```

**Comment by baconYao** [comment]
After discussing with Stanley, here are some things need to be fixed

- New folder called `API` and put `Jira` under it.
- Replace the partner's epic with dummy one.
- Empty line at the end in each file

File: API/Jira/apis/base.py [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+                response.raise_for_status()
+        except Exception as e:
+            logger.error(e)
+            logger.error('*' * 50)
+            logger.error(payload)
+            logger.error('*' * 50)
+        finally:
+            return response
+
+    def update_epic(self, epic='Dell', issues_id=[], **kwargs):
```
Comment (stanley31huang) [inline API/Jira/apis/base.py:76]: need to remove Dell, HP and Lenovo pattern

File: API/Jira/tests/test_base_api.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        dev_jira_project = {
+            'id': '10350',
+            'key': 'VS',
+            'name': 'Vic\'s Sandbox',
+            'issue_type': {
+                'Task': '10635',
+                'Story': '10634'
+            },
+            'epic': {
+                'Dell': '69110',
```
Comment (stanley31huang) [inline API/Jira/tests/test_base_api.py:31]: Per discussion, please remove Dell, HP and Lenovo pattern

**Review by stanley31huang — COMMENTED** [review]
Per discussion, please remove some vendor pattern from the scripts.

File: API/Jira/apis/base.py [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+        """ [Workaround] Update the epic of issues
+
+            Parameters:
+                epic {string}: Should field in the name of epic. Valid value
+                               should be 'Dell', 'HP' or 'Lenovo'
```
Comment (LiaoU3) [inline API/Jira/apis/base.py:81]: And here also.

**Review by stanley31huang — APPROVED** [review]
LGTM

**Pull request merged**